dc/s2n-quic-dc/src/path/secret/map/state.rs

```diff
@@ -119,30 +127,154 @@ impl State {
         let handshakes = self.requested_handshakes.pin();
         if handshakes.len() <= 6000 {
             handshakes.insert(peer);
+            self.subscriber()
+                .on_path_secret_map_background_handshake_requested(
+                    event::builder::PathSecretMapBackgroundHandshakeRequested {
+                        peer_address: SocketAddress::from(peer).into_event(),
+                    },
+                );
         }
     }
 
-    fn handle_unknown_secret_packet(&self, packet: &control::unknown_path_secret::Packet) {
+    fn handle_unknown_secret(
+        &self,
+        packet: &control::unknown_path_secret::Packet,
+        peer: &SocketAddress,
+    ) {
+        let peer_address = peer.into_event();
+
+        self.subscriber().on_unknown_path_secret_packet_received(
+            event::builder::UnknownPathSecretPacketReceived {
+                credential_id: packet.credential_id().into_event(),
+                peer_address,
+            },
+        );
+
         let Some(entry) = self.get_by_id(packet.credential_id()) else {
             return;
         };
+
         // Do not mark as live, this is lightly authenticated.
 
         // ensure the packet is authentic
         if packet
             .authenticate(&entry.sender().stateless_reset)
             .is_none()
         {
+            self.subscriber().on_unknown_path_secret_packet_rejected(
+                event::builder::UnknownPathSecretPacketRejected {
+                    credential_id: packet.credential_id().into_event(),
+                    peer_address,
+                },
+            );
+
             return;
         }
 
-        self.handled_control_packets.fetch_add(1, Ordering::Relaxed);
+        self.subscriber().on_unknown_path_secret_packet_accepted(
+            event::builder::UnknownPathSecretPacketAccepted {
+                credential_id: packet.credential_id().into_event(),
+                peer_address,
+            },
+        );
 
         // FIXME: More actively schedule a new handshake.
         // See comment on requested_handshakes for details.
         self.request_handshake(*entry.peer());
     }
 
+    fn handle_stale_key(&self, packet: &control::stale_key::Packet, peer: &SocketAddress) {
+        let peer_address = peer.into_event();
+
+        self.subscriber()
+            .on_stale_key_packet_received(event::builder::StaleKeyPacketReceived {
+                credential_id: packet.credential_id().into_event(),
+                peer_address,
+            });
+
+        let Some(entry) = self.ids.get_by_key(packet.credential_id()) else {
+            // If we get a control packet we don't have a registered path secret for, ignore the
+            // packet.
+            return;
+        };
+
+        let key = entry.control_secret();
+
+        let Some(packet) = packet.authenticate(&key) else {
+            self.subscriber().on_stale_key_packet_rejected(
```

Comment:
It seems like we ought to have a variant of rejected for failed auth and for just unknown?

Reply:
yeah i went back and forth on that. i currently don't have one for unknown - we don't emit anything right now. but probably worth doing.